app.js

```diff
@@ -0,0 +1,23 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
+require('env2')('./config.env');
+const reportRoutes = require('./routes/reportRoutes');
+
+mongoose.connect(process.env.MONGO_URL);
+
+const app = express();
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+app.get('/', (req, res) => {
+  res.send('Hello world');
+});
+
+app.use('/api', reportRoutes);
+
+const port = 5000 || process.env.PORT;
+const host = 'localhost' || process.env.HOST;
+app.listen(port, () => {
```

Comment:
Hmm don't you want to run your server when you database actually connected ?
so `mongoose.connect` can have event listeners, for example on error and on connected.

check [this out](http://theholmesoffice.com/mongoose-connection-best-practice/#managingthemongooseconnection)
so i would highly suggest to add the error event listener and the `connected` listener for when the database has connected and then you would want your server to connect after the db

Reply:
Great, done!!